test/php/gutenberg-coding-standards/Gutenberg/Sniffs/Commenting/FunctionCommentSinceTagSniff.php

```diff
@@ -48,90 +91,444 @@ public function process( File $phpcsFile, $stackPtr ) {
 			return;
 		}
 
-		$tokens        = $phpcsFile->getTokens();
-		$function_name = $phpcsFile->getDeclarationName( $stackPtr );
+		$tokens = $phpcsFile->getTokens();
+		$token  = $tokens[ $stackPtr ];
+
+		if ( 'T_FUNCTION' === $token['type'] ) {
+			$this->process_function_token( $phpcsFile, $stackPtr );
+			return;
+		}
+
+		if ( isset( static::$oo_tokens[ $token['code'] ] ) ) {
+			$this->process_oo_token( $phpcsFile, $stackPtr );
+			return;
+		}
 
-		$wrapping_tokens_to_check = array(
-			T_CLASS,
-			T_INTERFACE,
-			T_TRAIT,
+		if ( 'T_STRING' === $token['type'] ) {
+			$this->process_hook( $phpcsFile, $stackPtr );
+			return;
+		}
+
+		if ( 'T_VARIABLE' === $token['type'] && Scopes::isOOProperty( $phpcsFile, $stackPtr ) ) {
+			$this->process_property_token( $phpcsFile, $stackPtr );
+		}
+	}
+
+	/**
+	 * Processes a token representing a function call that invokes a WordPress hook,
+	 * checking for a missing `@since` tag in its docblock.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position of the hook token in the stack.
+	 */
+	protected function process_hook( File $phpcs_file, $stack_pointer ) {
```

Comment:
The code below triggers a false positive `MissingHookSinceTag` error (there might be other false positive cases if there are other cases where the name of one of the hook functions can be used as a string and not in a function call):

```php
class do_action {
	
}
```

This is likely not a problem for this sniff in the context of the Gutenberg project, but something to be aware of if we move this sniff to WPCS in the future. Not sure if you want to address it now or not.

Reply:
Thanks for addressing this as well. The `is_function_call()` method seems to be working correctly as far as I can check for the purposes needed for this sniff.